presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -255,10 +254,10 @@ public Object evaluate()
         return visitor.process(expression, new NoPagePositionContext());
     }
 
-    public Object evaluate(int position, Page page)
+    public Object evaluate(SymbolResolver inputs)
     {
         checkState(!optimize, "evaluate(int, Page) not allowed for optimizer");
-        return visitor.process(expression, new SinglePagePositionContext(position, page));
+        return visitor.process(expression, inputs);
     }
 
     public Object optimize(SymbolResolver inputs)
```

Comment:
Might as well change the error message in this `checkState`, or if you plan to merge `evaluate` and `optimize`, make sure the in the error message it says the right word. 😝